builtin/providers/aws/resource_aws_api_gateway_method_response.go

```diff
@@ -0,0 +1,207 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayMethodResponse() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayMethodResponseCreate,
+		Read:   resourceAwsApiGatewayMethodResponseRead,
+		Update: resourceAwsApiGatewayMethodResponseUpdate,
+		Delete: resourceAwsApiGatewayMethodResponseDelete,
+
+		Schema: map[string]*schema.Schema{
+			"api_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"http_method": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"status_code": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"response_models": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     schema.TypeString,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayMethodResponseCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	models := make(map[string]string)
+	for k, v := range d.Get("response_models").(map[string]interface{}) {
+		models[k] = v.(string)
+	}
+
+	_, err := conn.PutMethodResponse(&apigateway.PutMethodResponseInput{
+		HttpMethod:         aws.String(d.Get("http_method").(string)),
+		ResourceId:         aws.String(d.Get("resource_id").(string)),
+		RestApiId:          aws.String(d.Get("api_id").(string)),
+		StatusCode:         aws.String(d.Get("status_code").(string)),
+		ResponseModels:     aws.StringMap(models),
+		ResponseParameters: nil,
+	})
+	if err != nil {
+		return fmt.Errorf("Error creating API Gateway Method Response: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s-%s-%s-%s", d.Get("api_id").(string), d.Get("resource_id").(string), d.Get("http_method").(string), d.Get("status_code").(string)))
+	log.Printf("[DEBUG] API Gateway Method ID: %s", d.Id())
+
+	return nil
+}
+
+func resourceAwsApiGatewayMethodResponseRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[DEBUG] Reading API Gateway Method %s", d.Id())
+	out, err := conn.GetMethodResponse(&apigateway.GetMethodResponseInput{
+		HttpMethod: aws.String(d.Get("http_method").(string)),
+		ResourceId: aws.String(d.Get("resource_id").(string)),
+		RestApiId:  aws.String(d.Get("api_id").(string)),
+		StatusCode: aws.String(d.Get("status_code").(string)),
+	})
+	if err != nil {
+		return err
```

Comment:
We need to add guarding against deletion outside of Terraform here, like you did nicely for many other resources.